src/components/DateTimePicker/DateTimePicker.jsx

```diff
@@ -0,0 +1,702 @@
+  );
+
+  const onFieldClick = () => {
+    setIsExpanded(!isExpanded);
+  };
+
+  let focusOnFirstField = true;
+  /* istanbul ignore next */
+  const onDatePickerChange = range => {
+    if (range.length > 1) {
+      focusOnFirstField = !focusOnFirstField;
+    }
+    if (focusOnFirstField && datePickerRef.current.inputField) {
+      datePickerRef.current.inputField.focus();
+    } else {
+      datePickerRef.current.toInputField.focus();
+    }
+
+    const newAbsolute = { ...absoluteValue };
+    [newAbsolute.start] = range;
+    newAbsolute.startDate = moment(newAbsolute.start).format('MM/DD/YYYY');
+    newAbsolute.end = range[range.length - 1];
+    newAbsolute.endDate = moment(newAbsolute.end).format('MM/DD/YYYY');
+    setAbsoluteValue(newAbsolute);
+  };
+
+  const onDatePickerClose = (range, single, flatpickr) => {
+    // force it to stay open
+    /* istanbul ignore next */
+    if (flatpickr) {
+      flatpickr.open();
+    }
+  };
+
+  const onCustomRangeChange = kind => {
+    setCustomRangeKind(kind);
+  };
+
+  const toggleIsCustomRange = () => {
+    setIsCustomRange(!isCustomRange);
+    setSelectedPreset(null);
+  };
+
+  const onPresetClick = preset => {
+    setSelectedPreset(preset.offset);
+    renderValue(preset);
+  };
+
+  const parseDefaultValue = () => {
+    if (defaultValue !== null) {
+      if (defaultValue.hasOwnProperty('offset')) {
+        // preset
+        onPresetClick(defaultValue);
+      }
+      if (defaultValue.hasOwnProperty('lastNumber')) {
+        // relative
+        setIsCustomRange(true);
+        setCustomRangeKind(PICKER_KINDS.RELATIVE);
+        setRelativeValue(defaultValue);
+      }
+
+      if (defaultValue.hasOwnProperty('startDate')) {
+        // absolute
+        const absolute = { ...defaultValue };
+        setIsCustomRange(true);
+        setCustomRangeKind(PICKER_KINDS.ABSOLUTE);
+        if (!absolute.hasOwnProperty('start')) {
+          absolute.start = moment(absolute.startDate).valueOf();
+        }
+        if (!absolute.hasOwnProperty('end')) {
+          absolute.end = moment(absolute.endDate).valueOf();
+        }
+        setAbsoluteValue(absolute);
+      }
+    } else {
+      onPresetClick(presets[0]);
+    }
+  };
+
+  useEffect(
+    () => {
+      if (defaultValue || humanValue === null) {
+        parseDefaultValue();
+      }
+    },
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+    [defaultValue]
+  );
+
+  const onCancelClick = () => {
+    setIsExpanded(false);
+    parseDefaultValue();
+    if (onCancel) {
+      onCancel();
+    }
+  };
+
+  const onApplyClick = () => {
+    setIsExpanded(false);
+    const value = renderValue();
+    if (onApply) {
+      onApply(value);
+    }
+  };
+
+  const getIntervalValue = () => {
+    if (currentValue) {
+      if (currentValue.kind === PICKER_KINDS.PRESET) {
+        return `${moment()
+          .subtract(currentValue.preset.offset, 'minutes')
+          .format('YYYY-MM-DD HH:mm')} to Now`;
```

Comment:
not sure how this is used but if it's displayed then we need translation and custom date/time mask